compiler/rustc_mir_transform/src/gvn.rs

```diff
@@ -406,6 +429,28 @@ impl<'body, 'tcx> VnState<'body, 'tcx> {
                     };
                     let ptr_imm = Immediate::new_pointer_with_meta(data, meta, &self.ecx);
                     ImmTy::from_immediate(ptr_imm, ty).into()
+                } else if matches!(kind, AggregateTy::Array) {
+                    if ty.layout.size().bytes() <= STACK_THRESHOLD {
+                        return None;
+                    }
```

Comment:
For this case, should we attempt to fallback to the next branch?

Reply:
I mean the `matches!(ty.abi, Abi::Scalar(..) | Abi::ScalarPair(..))` branch.

Reply:
But `ty.abi` of array is not `Abi::Scalar*` though. And with STACK_THRESHOLD, I fallback to default behavior on master branch. Also I committed 4fb498ec278d3ac22095c8e58841594c3b58923d. 
But I don't get what the differences between `ecx.intern_with_temp_alloc` and` ecx.alloc`, in the case of big array (`sizeof(array) > stack_size`).

Reply:
Alright. You convinced me about this is llvm's job. Reverted the change.

Merging two branches should be easy.